WalletWasabi/BitcoinCore/Rpc/RpcParser.cs

```diff
@@ -27,53 +28,63 @@ public static RpcPubkeyType ConvertPubkeyType(string pubKeyType)
+			if (!parsed.TryGetProperty("result", out JsonElement blockInfoJson))
+			{
+				blockInfoJson = parsed;
+			}
+
+			string? previousBlockHash = null;
+			if (blockInfoJson.TryGetProperty("previousblockhash", out var prevBlockHashJson))
+			{
+				previousBlockHash = prevBlockHashJson.GetString();
+			}
+
 			var transaction = new List<VerboseTransactionInfo>();
 
 			var blockInfo = new VerboseBlockInfo(
-				hash: uint256.Parse(blockInfoJson.Value<string>("hash")),
-				prevBlockHash: previousBlockHash is { } ? uint256.Parse(previousBlockHash) : uint256.Zero,
-				confirmations: blockInfoJson.Value<ulong>("confirmations"),
-				height: blockInfoJson.Value<ulong>("height"),
-				blockTime: Utils.UnixTimeToDateTime(blockInfoJson.Value<uint>("time")),
-				transactions: transaction
-			);
+				hash: uint256.Parse(blockInfoJson.GetProperty("hash").GetString()),
+				prevBlockHash: previousBlockHash is not null ? uint256.Parse(previousBlockHash) : uint256.Zero,
+				confirmations: blockInfoJson.GetProperty("confirmations").GetUInt64(),
+				height: blockInfoJson.GetProperty("height").GetUInt64(),
+				blockTime: Utils.UnixTimeToDateTime(blockInfoJson.GetProperty("time").GetUInt32()),
+				transactions: transaction);
 
-			foreach (var txJson in blockInfoJson["tx"])
+			var array = blockInfoJson.GetProperty("tx").EnumerateArray().ToArray();
+			for (uint i = 0; i < array.Length; i++)
 			{
+				var txJson = array[i];
 				var inputs = new List<VerboseInputInfo>();
 				var outputs = new List<VerboseOutputInfo>();
-				var tx = new VerboseTransactionInfo(uint256.Parse(txJson.Value<string>("txid")), inputs, outputs);
+				var txBlockInfo = new TransactionBlockInfo(blockInfo.Hash, blockInfo.BlockTime, i);
+				var tx = new VerboseTransactionInfo(txBlockInfo, uint256.Parse(txJson.GetProperty("txid").GetString()), inputs, outputs);
 
-				foreach (var txinJson in txJson["vin"])
+				foreach (var txinJson in txJson.GetProperty("vin").EnumerateArray())
 				{
 					VerboseInputInfo input;
-					if (txinJson["coinbase"] is { })
+					if (txinJson.TryGetProperty("coinbase", out JsonElement cb))
 					{
-						input = new VerboseInputInfo(txinJson["coinbase"].Value<string>());
+						input = new VerboseInputInfo(cb.GetString() ?? "");
 					}
 					else
 					{
 						input = new VerboseInputInfo(
-							outPoint: new OutPoint(uint256.Parse(txinJson.Value<string>("txid")), txinJson.Value<uint>("vout")),
+							outPoint: new OutPoint(uint256.Parse(txinJson.GetProperty("txid").GetString()), txinJson.GetProperty("vout").GetUInt32()),
 							prevOutput: new VerboseOutputInfo(
-								value: Money.Coins(txinJson["prevout"].Value<decimal>("value")),
-								scriptPubKey: Script.FromHex(txinJson["prevout"]["scriptPubKey"].Value<string>("hex")),
-								pubkeyType: txinJson["prevout"]["scriptPubKey"].Value<string>("type"))
-						);
+								value: Money.Coins(txinJson.GetProperty("prevout").GetProperty("value").GetDecimal()),
```

Comment:
`txinJson.GetProperty("prevout")` is called multiple times. One can also add a variable to re-use it ([JsonElement.GetProperty(string)](https://source.dot.net/#System.Text.Json/System/Text/Json/Document/JsonElement.cs,eaf74e3f1b1678d7) is nontrivial operation but it would still be a micro-optimization.)

Reply:
Done!